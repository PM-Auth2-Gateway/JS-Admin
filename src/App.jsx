import React from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import styles from './App.module.scss';

import Navigation from './components/Navigation/Navigation';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import ApplicationsPage from './pages/ApplicationsPage/ApplicationsPage';
import SingleApplicationPage from './pages/SingleApplicationPage/SingleApplicationPage';

function App() {
  return (
    <Router>
      <div className={styles.App}>
        <Navigation />
      </div>
      <div className={styles.container}>
        <Switch>
          <PrivateRoute exact path='/applications/:appId' component={SingleApplicationPage} />
          <PrivateRoute exact path='/applications' component={ApplicationsPage} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
