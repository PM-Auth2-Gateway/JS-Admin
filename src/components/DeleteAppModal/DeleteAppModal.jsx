import React, { useState } from 'react';
import { Button, Modal } from 'react-bootstrap';

const DeleteAppModal = ({ onDelete }) => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const onSubmit = (values) => {
    onDelete();
    handleClose();
  };

  return (
    <React.Fragment>
      <Button data-test={'open-modal'} variant='danger' onClick={handleShow}>
        Delete
      </Button>

      <Modal show={show} onHide={handleClose} centered>
        <Modal.Header closeButton>
          <Modal.Title>Delete Application</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          Are you sure you want to delete this application?
        </Modal.Body>

        <Modal.Footer>
          <Button
            data-test={'close-modal'}
            variant='outline-secondary'
            onClick={handleClose}
          >
            Cancel
          </Button>
          <Button
            data-test={'submit-modal'}
            variant='primary'
            onClick={onSubmit}
          >
            Delete
          </Button>
        </Modal.Footer>
      </Modal>
    </React.Fragment>
  );
};

export default DeleteAppModal;
