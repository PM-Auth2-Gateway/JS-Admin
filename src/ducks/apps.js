import config from '../config';
import AppsApiService from '../services/AppsApiService';

const LOAD_ALL = 'apps/all/load';
const ADD_ALL = 'apps/all/add';
const DELETE_BY_ID_ALL = 'apps/all/delete_by_id';
const CLEAR_ALL = 'apps/all/clear';

const SET_CURRENT = 'apps/current/set';
const REMOVE_CURRENT = 'apps/current/delete';

const SET_ERROR = 'apps/error/set';

export const loadAllApps = () => (dispatch) => {
  AppsApiService.get(config.authToken)
    .then((response) => {
      dispatch(loadAll(response.data));
    })
    .catch((error) => {
      dispatch(setError(error));
    });
};

export const addApp = (name) => (dispatch) => {
  debugger;
  AppsApiService.post(config.authToken, { name })
    .then((response) => {
      dispatch(addAll(response.data));
    })
    .catch((error) => {
      dispatch(setError(error));
    });
};

export const deleteAppById = (id) => (dispatch) => {
  AppsApiService.deleteById(config.authToken, { id })
    .then((response) => {
      dispatch(deleteByIdAll(id));
    })
    .catch((error) => {
      dispatch(setError(error));
    });
};

export const clearAllApps = () => (dispatch) => {
  dispatch(clearAll());
};

export const loadCurrentApp = (id) => (dispatch) => {
  AppsApiService.getById(config.authToken, { id })
    .then((response) => {
      dispatch(setCurrent(response.data));
    })
    .catch((error) => {
      dispatch(setError(error));
    });
};

export const removeCurrentApp = () => (dispatch) => {
  dispatch(removeCurrent());
};

export const setAppError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const addAll = (app) => ({
  type: ADD_ALL,
  payload: app,
});

const deleteByIdAll = (id) => ({
  type: DELETE_BY_ID_ALL,
  payload: id,
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setCurrent = (app) => ({
  type: SET_CURRENT,
  payload: app,
});

const removeCurrent = () => ({
  type: REMOVE_CURRENT,
});

const setError = (msg) => ({
  type: SET_CURRENT,
  payload: msg,
});

const initialState = {
  all: [],
  current: {},
  error: {},
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        all: [...state.all, ...action.payload],
      };
    case ADD_ALL:
      return {
        ...state,
        all: [...state.all, action.payload],
      };
    case DELETE_BY_ID_ALL:
      return {
        ...state,
        all: state.all.filter((app) => app.id !== action.payload.id),
      };
    case CLEAR_ALL:
      return {
        ...state,
        all: [],
      };
    case SET_CURRENT:
      return {
        ...state,
        current: action.payload,
      };
    case REMOVE_CURRENT:
      return {
        ...state,
        current: null,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
