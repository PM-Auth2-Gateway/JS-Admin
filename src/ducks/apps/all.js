import AppsApiService from '../../services/AppsApiService';

const LOAD_ALL = 'apps/all/load';
const ADD_ALL = 'apps/all/add';
const CLEAR_ALL = 'apps/all/clear';

const SET_LOADING = 'apps/all/loading/set';
const SET_ERROR = 'apps/all/error/set';

export const loadAllApps = () => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.get();

    dispatch(loadAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const addApp = ({ name }) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.post({ name });

    dispatch(addAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const clearAllApps = () => (dispatch) => {
  dispatch(clearAll());
};

export const setAppError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const addAll = (app) => ({
  type: ADD_ALL,
  payload: app,
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: [],
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        value: [...state.value, ...action.payload],
      };
    case ADD_ALL:
      return {
        ...state,
        value: [...state.value, action.payload],
      };
    case CLEAR_ALL:
      return {
        ...state,
        value: [],
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
