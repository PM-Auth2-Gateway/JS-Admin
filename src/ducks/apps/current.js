import config from '../../config';
import AppsApiService from '../../services/AppsApiService';

const SET_CURRENT = 'apps/current/set';
const UPDATE_CURRENT = 'app/current/update';
const DELETE_CURRENT = 'apps/current/delete';
const REMOVE_CURRENT = 'apps/current/remove';

const SET_LOADING = 'apps/current/loading/set';
const SET_ERROR = 'apps/current/error/set';

export const loadCurrentApp = (id) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.getById(config.authToken, { id });

    dispatch(setCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateCurrentApp = (data) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.updateById(config.authToken, data);

    dispatch(updateCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const deleteCurrentApp = (id) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    await AppsApiService.deleteById(config.authToken, { id });

    dispatch(deleteCurrent());
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const removeCurrentApp = () => (dispatch) => {
  dispatch(removeCurrent());
};

const setCurrent = (app) => ({
  type: SET_CURRENT,
  payload: app,
});

const updateCurrent = (data) => ({
  type: UPDATE_CURRENT,
  payload: data,
});

const deleteCurrent = (id) => ({
  type: DELETE_CURRENT,
  payload: id,
});

const removeCurrent = () => ({
  type: REMOVE_CURRENT,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: {},
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case SET_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case DELETE_CURRENT:
      return {
        ...state,
        value: {},
      };
    case REMOVE_CURRENT:
      return {
        ...state,
        value: {},
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
