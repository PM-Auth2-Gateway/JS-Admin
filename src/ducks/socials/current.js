import SocialsApiService from '../../services/SocialsApiService';

const SET_CURRENT = 'socials/current/set';
const UPDATE_CURRENT = 'socials/current/update';
const REMOVE_CURRENT = 'socials/current/remove';

const SET_LOADING = 'socials/current/loading/set';
const SET_ERROR = 'socials/current/error/set';

export const loadCurrentSocial = (appId, id) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.getById(appId, id);

    dispatch(setCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateCurrentSocial = (
  appId,
  id,
  { client_id, scope, secret_key }
) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.updateById(appId, id, {
      client_id,
      scope,
      secret_key,
    });

    dispatch(updateCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const removeCurrentSocial = () => (dispatch) => {
  dispatch(removeCurrent());
};

const setCurrent = (app) => ({
  type: SET_CURRENT,
  payload: app,
});

const updateCurrent = (data) => ({
  type: UPDATE_CURRENT,
  payload: data,
});

const removeCurrent = () => ({
  type: REMOVE_CURRENT,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: {},
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case SET_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case REMOVE_CURRENT:
      return {
        ...state,
        value: {},
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
