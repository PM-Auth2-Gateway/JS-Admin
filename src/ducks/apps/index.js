import { combineReducers } from 'redux';

import { reducer as appsAllReducer } from './all';
import { reducer as appsCurrentReducer } from './current';

export const reducer = combineReducers({
  all: appsAllReducer,
  current: appsCurrentReducer,
});
