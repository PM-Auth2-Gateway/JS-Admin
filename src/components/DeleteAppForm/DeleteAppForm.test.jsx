import { shallow } from 'enzyme';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { useAppContext } from '../../contexts/App.context';
import { deleteCurrentApp } from '../../ducks/apps/current';
import DeleteAppModal from '../DeleteAppModal/DeleteAppModal';

import DeleteAppFrom from './DeleteAppForm';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(),
}));

jest.mock('../../contexts/App.context', () => ({
  useAppContext: jest.fn(),
}));

jest.mock('../../ducks/apps/current', () => ({
  deleteCurrentApp: jest.fn(),
}));

describe('DeleteAppForm Component', () => {
  const dispatch = jest.fn();
  const historyPush = jest.fn();
  const action = jest.fn;

  beforeAll(() => {
    deleteCurrentApp.mockReturnValue(action);
  });

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);

    useAppContext.mockReturnValue({
      appId: 1,
    });
    useHistory.mockReturnValue({
      push: historyPush,
    });
  });

  it('should render component', () => {
    const component = shallow(<DeleteAppFrom />);

    expect(component).toMatchSnapshot();
  });

  it('should call dispatch and redirect to /applications', async () => {
    const component = shallow(<DeleteAppFrom />);

    const deleteModal = component.find(DeleteAppModal);

    await deleteModal.prop('onDelete')();

    expect(dispatch).toBeCalledTimes(1);
    expect(historyPush).toBeCalledWith('/applications');
  });
});
