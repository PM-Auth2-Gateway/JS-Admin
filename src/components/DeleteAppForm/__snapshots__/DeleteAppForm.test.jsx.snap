// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAppForm Component should render component 1`] = `
<Fragment>
  <h3
    className="my-3"
  >
    Danger zone
  </h3>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "render": [Function],
      }
    }
    variant="danger"
  >
    <Row
      noGutters={false}
    >
      <Col>
        <h5>
          Delete this application
        </h5>
        <p>
          All your apps using this client will stop working.
        </p>
      </Col>
      <Col
        className="align-self-center"
        xs="auto"
      >
        <DeleteAppModal
          onDelete={[Function]}
        />
      </Col>
    </Row>
  </Alert>
</Fragment>
`;
